Version 0.6-alpha broke role-based authorization that contains global roles only

If jenkins.yaml contains role-based authorization configuration with global roles only
```
jenkins:
  securityRealm:
    local:
      allowsSignup: false
      users:
        - id: "admin"
          password: 12345

  # Configure role-based authorization strategy
  authorizationStrategy:
    roleStrategy:
      roles:
        global:
          - name: "admin"
            description: "Jenkins administrators"
            permissions:
              - "Overall/Administer"
            assignments:
              - "admin"          
```

Then this setup doesn't work in 0.6-alpha (however it worked in 0.4-alpha)
In 0.6-alpha Jenkins fails to start with exception
```
Caused by: java.lang.NullPointerException
jenkins    |    at org.jenkinsci.plugins.casc.integrations.rolebasedauth.RoleBasedAuthorizationStrategyConfigurator.retrieveRoleMap(RoleBasedAuthorizationStrategyConfigurator.java:71)
jenkins    |    at org.jenkinsci.plugins.casc.integrations.rolebasedauth.RoleBasedAuthorizationStrategyConfigurator.configure(RoleBasedAuthorizationStrategyConfigurator.java:61)
jenkins    |    at org.jenkinsci.plugins.casc.integrations.rolebasedauth.RoleBasedAuthorizationStrategyConfigurator.configure(RoleBasedAuthorizationStrategyConfigurator.java:33)
jenkins    |    at org.jenkinsci.plugins.casc.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:81)
jenkins    |    at org.jenkinsci.plugins.casc.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:34)
jenkins    |    at org.jenkinsci.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:188)
jenkins    |    at org.jenkinsci.plugins.casc.JenkinsConfigurator.configure(JenkinsConfigurator.java:44)
jenkins    |    at org.jenkinsci.plugins.casc.JenkinsConfigurator.configure(JenkinsConfigurator.java:24)
jenkins    |    at org.jenkinsci.plugins.casc.ConfigurationAsCode.configureWith(ConfigurationAsCode.java:335)
jenkins    |    at org.jenkinsci.plugins.casc.ConfigurationAsCode.configureWith(ConfigurationAsCode.java:280)
jenkins    |    at org.jenkinsci.plugins.casc.ConfigurationAsCode.configure(ConfigurationAsCode.java:252)
jenkins    |    at org.jenkinsci.plugins.casc.ConfigurationAsCode.configure(ConfigurationAsCode.java:155)
jenkins    |    at org.jenkinsci.plugins.casc.ConfigurationAsCode.init(ConfigurationAsCode.java:134)

```

As far as I can tell it is caused by this change:
https://github.com/jenkinsci/configuration-as-code-plugin/commit/f7cfe0d2a32c6f31d76dea5322c45023d7eefc4a#diff-c4e07dfa3bcbf094062d1295a61d5ffbL70

Missing `items` or `agents` field causes NPE because new code tries to convert them to sequence without checking for their existence.

Specifying  empty project and agent roles  like this solves problem:

```
jenkins:
  securityRealm:
    local:
      allowsSignup: false
      users:
        - id: "admin"
          password: 12345

  # Configure role-based authorization strategy
  authorizationStrategy:
    roleStrategy:
      roles:
        items: []
        agents: []
        global:
          - name: "admin"
            description: "Jenkins administrators"
            permissions:
              - "Overall/Administer"
            assignments:
              - "admin"          
```

--- a/src/main/java/org/jenkinsci/plugins/casc/integrations/rolebasedauth/RoleBasedAuthorizationStrategyConfigurator.java
+++ b/src/main/java/org/jenkinsci/plugins/casc/integrations/rolebasedauth/RoleBasedAuthorizationStrategyConfigurator.java
@@ -68,15 +68,15 @@ public class RoleBasedAuthorizationStrategyConfigurator extends Configurator<Rol
     @Nonnull
     private static RoleMap retrieveRoleMap(@Nonnull CNode config, @Nonnull String name, Configurator<RoleDefinition> configurator) throws ConfiguratorException {
         Mapping map = config.asMapping();
-        final Sequence c = map.get(name).asSequence();
+        final CNode c = map.get(name);
 
         TreeMap<Role, Set<String>> resMap = new TreeMap<>();
-        if (c == null) {
+        if (c == null || c.asSequence() == null) {
             // we cannot return emptyMap here due to the Role Strategy code
             return new RoleMap(resMap);
         }
 
-        for (CNode entry : c) {
+        for (CNode entry : c.asSequence()) {
             RoleDefinition definition = configurator.configure(entry);
             resMap.put(definition.getRole(), definition.getAssignments());
         }

--- a/src/test/java/org/jenkinsci/plugins/casc/integrations/rolebasedauth/RoleStrategyTest.java
+++ b/src/test/java/org/jenkinsci/plugins/casc/integrations/rolebasedauth/RoleStrategyTest.java
@@ -104,4 +104,17 @@ public class RoleStrategyTest {
         // Same user still cannot build on agent2
         assertHasNoPermission(user1, agent2, Computer.BUILD);
     }
+
+    @Test
+    @Issue("Issue #214")
+    @ConfiguredWithCode("RoleStrategy2.yml")
+    public void shouldHandleNullItemsAndAgentsCorrectly() throws Exception {
+        AuthorizationStrategy s = j.jenkins.getAuthorizationStrategy();
+        assertThat("Authorization Strategy has been read incorrectly",
+            s, instanceOf(RoleBasedAuthorizationStrategy.class));
+        RoleBasedAuthorizationStrategy rbas = (RoleBasedAuthorizationStrategy) s;
+
+        Map<Role, Set<String>> globalRoles = rbas.getGrantedRoles(RoleBasedAuthorizationStrategy.GLOBAL);
+        assertThat(globalRoles.size(), equalTo(2));
+    }
 }

--- a/src/test/resources/org/jenkinsci/plugins/casc/integrations/rolebasedauth/RoleStrategy2.yml
+++ b/src/test/resources/org/jenkinsci/plugins/casc/integrations/rolebasedauth/RoleStrategy2.yml
@@ -1,0 +1,43 @@
+jenkins:
+
+  authorizationStrategy:
+    roleStrategy:
+      roles:
+        global:
+          - name: "admin"
+            description: "Jenkins administrators"
+            permissions:
+              - "Overall/Administer"
+            assignments:
+              - "admin"
+          - name: "readonly"
+            description: "Read-only users"
+            permissions:
+              - "Overall/Read"
+              - "Job/Read"
+            assignments:
+              - "authenticated"
+
+  # System for test
+  securityRealm:
+    local:
+      allowsSignup: false
+      users:
+        - id: "admin"
+          password: "1234"
+        - id: "user1"
+          password: ""
+
+  nodes:
+    - dumb:
+        mode: NORMAL
+        name: "agent1"
+        remoteFS: "/home/user1"
+        launcher:
+          jnlp:
+    - dumb:
+        mode: NORMAL
+        name: "agent2"
+        remoteFS: "/home/user1"
+        launcher:
+          jnlp:

